### ISSUE TITLE
Missing Typing for Loss Functions Causes Poor Code Completion in VSCode

### ISSUE THREAD

**Issue by Julfried** [pytorch/pytorch#146831]
### 🐛 Describe the bug

Currently, all loss functions in torch.nn.modules.loss return a Tensor, but VSCode's IntelliSense and static type checkers (like Pyright) cannot infer this correctly. This makes code completion and function signatures less helpful when working with loss functions.

For example, in VSCode:

```python
import torch

loss_fn = torch.nn.CrossEntropyLoss()

loss = loss_fn(torch.tensor([0.1, 0.9]), torch.tensor([1.0, 0.0])) # Expected: Tensor, but loss is infered as Any
```

Because the return type is not properly inferred, IntelliSense does not suggest useful Tensor methods when working with loss, making development inconvenient.

### Versions

Versions of relevant libraries:
[pip3] numpy==2.1.3
[pip3] numpydoc==1.8.0
[pip3] torch==2.5.1
[conda] numpy                     2.1.3                    pypi_0    pypi
[conda] numpydoc                  1.8.0                    pypi_0    pypi
[conda] torch                     2.5.1                    pypi_0    pypi
